hpedockerplugin/synchronization.py

```diff
@@ -0,0 +1,39 @@
+import inspect
+import json
+
+from oslo_log import log as logging
+
+import exception
+
+LOG = logging.getLogger(__name__)
+
+
+def synchronized(lock_name):
+    def _synchronized(f):
+        def _wrapped(*a, **k):
+            call_args = inspect.getcallargs(f, *a, **k)
+            call_args['f_name'] = f.__name__
+            lck_name = lock_name.format(**call_args)
+            lock_acquired = False
+            plugin = a[0]
+            try:
+                plugin._etcd.try_lock_volname(lck_name)
+                lock_acquired = True
+                LOG.info("Lock acquired: [lock-name=%s]" % lck_name)
+                return f(*a, **k)
+            except exception.HPEPluginLockFailed:
+                LOG.exception('Failed to acquire lock: [lock-name=%(name)s]',
```

Comment:
Same as previous comment. Please add function name as part of logging.

Reply:
@wdurairaj This has been addressed in my second commit. Please have a look.